notebooks/dqc.ipynb

```diff
@@ -0,0 +1,2357 @@
+      "Column date            ->    0 nans\n",
+      "Column date_block_num  ->    0 nans\n",
+      "Column shop_id         ->    0 nans\n",
+      "Column item_id         ->    0 nans\n",
+      "Column item_price      ->    0 nans\n",
+      "Column item_cnt_day    ->    0 nans\n"
+     ]
+    }
+   ],
+   "source": [
+    "template = 'Column {:15} -> {:4} nans'\n",
+    "for col in sales.columns:\n",
+    "    print(template.format(col, sales[col].isna().sum()))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>date</th>\n",
+       "      <th>date_block_num</th>\n",
+       "      <th>shop_id</th>\n",
+       "      <th>item_id</th>\n",
+       "      <th>item_price</th>\n",
+       "      <th>item_cnt_day</th>\n",
+       "      <th>item_name</th>\n",
+       "      <th>item_category_id</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>15.05.2013</td>\n",
+       "      <td>4</td>\n",
+       "      <td>32</td>\n",
+       "      <td>2973</td>\n",
+       "      <td>-1.0</td>\n",
+       "      <td>1.0</td>\n",
+       "      <td>DmC Devil May Cry [PS3, русские субтитры]</td>\n",
+       "      <td>19</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "         date  date_block_num  shop_id  item_id  item_price  item_cnt_day  \\\n",
+       "0  15.05.2013               4       32     2973        -1.0           1.0   \n",
+       "\n",
+       "                                   item_name  item_category_id  \n",
+       "0  DmC Devil May Cry [PS3, русские субтитры]                19  "
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# see if there're any items with negative price\n",
+    "\n",
+    "sales[sales['item_price']<=0].merge(items, on='item_id', how='inner')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "found no `NaN`s, but found an item with negative price, which has to be removed later"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Looking for unresolved relations between tables\n",
```

Comment:
I think I did it a few cells below by left joining the `test` to `shops` and `items` respectively and then seeing if there are any `NaN`s (which would appear in case of values that are present only in `test`).